### ISSUE TITLE
Cannot import into typescript browser application

### ISSUE THREAD

**Issue by jonaskello** [cloudamqp/amqp-client.js#21]
I'm experimenting with using this package when building a browser application using typescript, react and webpack.

The problem I encounter is that I cannot import from this package because of the way the typings are exported.

Specifically this package has this in `package.json`:

```
  "name": "@cloudamqp/amqp-client",
  "version": "1.3.2",
  "description": "AMQP 0-9-1 client, both for browsers (WebSocket) and node (TCP Socket)",
  "type": "module",
  "main": "dist/amqp-client.cjs",
  "types": "dist/types/amqp-client.d.ts",
  "browser": "dist/amqp-websocket-client.mjs",
```

This seems to be correct usage of the `browser` field. The idea being that a browser application will get `dist/amqp-websocket-client.mjs` instead of `dist/amqp-client.cjs`.  However it does not work with typescript because there is only a single `types` field that can export types. So even if webpack will pick the `browser` field typescript will still pick the `types` field when compiling. For more info see [this issue](https://github.com/Microsoft/TypeScript/issues/29128) in the typscript repo.

Could we somehow rework the structure of the package so it will be possible to import the websocket client from a typescript app? Perhaps so it can be imported with an explicit path like `@cloudamqp/amqp-client/lib/amqp-websocket-client`?

If it helps, my [experimental app is here](https://github.com/jonaskello/amqp-websockets-example). It works but only becuase I copied all the typing files out from the node_modules folder and into the application.

**Comment by jonaskello** [comment]
Another issue is that it is not currently possible to import types from the package (if you eg. want to decorate your own function with types from this package). For example this does not work:

```ts
import { AMQPChannel } from "@cloudamqp/amqp-client";

function getChannel(): AMQPChannel {
 //
}
```

**Comment by carlhoerberg** [comment]
I think both issues are addressed in #22 now..

**Comment by carlhoerberg** [comment]
Fixed in 2.0.0
